src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -2142,10 +2142,24 @@ public string ToUpperInvariant()
         // Trims the whitespace from both ends of the string.  Whitespace is defined by
         // char.IsWhiteSpace.
         //
-        public string Trim() => TrimWhiteSpaceHelper(TrimType.Both);
+        public string Trim()
+        {
+            if (Length == 0 || (!char.IsWhiteSpace(_firstChar) && !char.IsWhiteSpace(this[^1])))
+            {
+                return this;
+            }
+            return TrimWhiteSpaceHelper(TrimType.Both);
+        }
 
         // Removes a set of characters from the beginning and end of this string.
-        public unsafe string Trim(char trimChar) => TrimHelper(&trimChar, 1, TrimType.Both);
+        public unsafe string Trim(char trimChar)
+        {
+            if (Length == 0 || (_firstChar != trimChar && this[^1] != trimChar))
+            {
+                return this;
+            }
+            return TrimHelper(&trimChar, 1, TrimType.Both);
```

Comment:
Not for this PR, but now that we'll be invoking these helpers only when we know there is something to be trimmed, I wonder whether employing an [IndexOfAnyExceptWhitespace](https://github.com/dotnet/runtime/issues/77959) would be more useful, i.e. whether there being any whitespace makes it more likely there's enough whitespace for vectorization to be a win.